### ISSUE TITLE
Why utilizing the 'question' column of the 'ArmelR/stack-exchange-instruction' dataset for gradient backpropagation?

### ISSUE THREAD

**Issue by HIT-cwh** [bigcode-project/starcoder#135]
Thank you for your valuable open-source contribution!

In instruction tuning stage, it seems that only the answer aligned with the instruction participates in the backpropagation process.  And [this code](https://github.com/bigcode-project/starcoder/blob/main/finetune/finetune.py#L190) seems to imply that the question part of the dataset is also involved in backpropagation of the gradient. Will this lead to better training results?

![image](https://github.com/bigcode-project/starcoder/assets/41630003/4569446f-c4f1-47e6-abe0-90f90766bda3)

**Comment by ArmelRandy** [comment]
Hi. Thank you for reaching out. In this implementation, we do not mask the part of the sequence which corresponds to the question. You can slightly modify the code to account for that. Both methods work reasonably well (masking and not masking) though masking seems to be the standard practice.
